### PR TITLE
Require user for collector inbox (PURCHASE-1914)

### PR DESCRIPTION
**Pull request by starsirius**
https://artsyproduct.atlassian.net/browse/PURCHASE-1914

Depending on https://github.com/artsy/force/pull/5780.

The current conversations app does not require a logged-in user, and that causes:

- visiting /user/conversations would 500
- visiting /user/conversations/:id would 404
- logging out from conversations page would visit the same route, so 500 or 404 similarly

This PR requires a user for the conversations app, and follows the existing behavior for logging out from other pages: rendering the login page with proper redirect param (to take user back to the original page after logging in).

## Before

conversations-not-requiring-user

## After

conversations-requiring-user

### CODE REVIEW

File: src/desktop/apps/artsy-v2/middleware/userRequiredMiddleware.tsx [hunk 1]
```diff
@@ -1,11 +1,19 @@
-import { getPageType } from "../utils/getPageType"
+import { match } from "path-to-regexp"
 
-const USER_REQUIRED_PAGES = ["identity-verification", "orders"]
+const USER_REQUIRED_ROUTES = [
+  "/identity-verification(.*)",
```
Comment (starsirius) [inline src/desktop/apps/artsy-v2/middleware/userRequiredMiddleware.tsx:4]: The current page type approach only uses the top level path for checking if a route should require a logged-in user or not. This makes it difficult to allow different requirements for nested routes. For example, we might require a logged-in user for /user/conversations but might not require that for /user/login (if this is the route we choose).

This commit switches to use regex for matching routes, and it's using [`path-to-regexp`](https://github.com/pillarjs/path-to-regexp), the same package Express.js uses for matching routes. Therefore, we can use the same regex syntax. I hope the regex matching would not be too expensive.

Comment (damassi) [reply src/desktop/apps/artsy-v2/middleware/userRequiredMiddleware.tsx:4]: @starsirius - i've been [switching our naive checks](https://github.com/artsy/force/pull/5588/files) over to `path-to-regex` latetly, so this is nice and timely 👍

Comment (damassi) [reply src/desktop/apps/artsy-v2/middleware/userRequiredMiddleware.tsx:4]: Speaking of which, @starsirius or @zephraph - I think the conversations route needs to be to that tracking exclude list in the link above. 

(We need to make this automatic (for lack of a more unified solution [as described here](https://artsyproduct.atlassian.net/browse/PLATFORM-2362)  and @mzikherman has done some prelim work for route lookup https://github.com/artsy/force/pull/5745. We should be able to piggy back on this code.

Comment (starsirius) [reply src/desktop/apps/artsy-v2/middleware/userRequiredMiddleware.tsx:4]: Thanks @damassi for raising that! I created a [ticket](https://artsyproduct.atlassian.net/browse/PURCHASE-1965) to double check that.

**Comment by starsirius** [comment]
Merging this! Thanks for review!

**Pull request merged**